includes/Admin/Admin_AJAX.php

```diff
@@ -278,6 +223,70 @@ public function run_checks() {
 		);
 	}
 
+
+	/**
+	 * Get the AJAX_Runner runner.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return AJAX_Runner
+	 */
+	protected function get_runner() {
+		$runner = Plugin_Request_Utility::get_runner();
+
+		if ( is_null( $runner ) ) {
+			Plugin_Request_Utility::initialize_runner();
+			$runner = Plugin_Request_Utility::get_runner();
+		}
+
+		// Make sure we are using the correct runner instance.
+		if ( ! ( $runner instanceof AJAX_Runner ) ) {
+			wp_send_json_error(
+				new WP_Error( 'invalid-runner', __( 'AJAX Runner was not initialized correctly.', 'plugin-check' ) ),
+				500
+			);
+		}
+
+		return $runner;
+	}
+
+	/**
+	 * Get array of checks to run.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	protected function get_checks() {
+		$checks = filter_input( INPUT_POST, 'checks', FILTER_DEFAULT, FILTER_FORCE_ARRAY );
+
+		return is_null( $checks ) ? array() : (array) $checks;
+	}
+
+	/**
+	 * Get requested plugin.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string
+	 */
+	protected function get_plugin() {
+		return filter_input( INPUT_POST, 'plugin', FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+	}
+
+	/**
+	 * Validate request by checking nonce.
+	 *
+	 * @since n.e.x.t
+	 */
+	protected function validate_request() {
+		$valid_request = $this->verify_request( filter_input( INPUT_POST, 'nonce', FILTER_SANITIZE_FULL_SPECIAL_CHARS ) );
+
+		if ( is_wp_error( $valid_request ) ) {
+			wp_send_json_error( $valid_request, 403 );
+		}
+	}
```

Comment:
See above. This method shouldn't be necessary at all. I would suggest we rather modify the `verify_request()` method to throw an exception instead of returning `WP_Error`. That way we can catch a potential exception with the same code that we would use also for the other methods, like `get_runner()` above.